pkg/gcp/machine_controller.go

```diff
@@ -75,11 +75,13 @@ func (ms *MachinePlugin) CreateMachine(ctx context.Context, req *driver.CreateMa
 		return nil, status.Error(codes.InvalidArgument, err.Error())
 	}
 
-	providerSpec, err := decodeProviderSpecAndSecret(req.MachineClass, req.Secret)
+	providerSpec, err := decodeProviderSpecAndValidateSecret(req.MachineClass, req.Secret)
```

Comment:
Since you are separating the decoding from validation then it would then be better to segregate it cleanly. What you have today is decoding of provider spec mixed with validation of secret but validation of provider spec is done separately.
So you have 3 choices:

_Option 1_
You should then have 3 functions:
`decodeProviderSpec`
`validateProviderSpec`
`validateSecret`

_Option 2_
jut keep 2 functions:
`decodeAndValidateProviderSpec`
`validateSecret`

_Option 3_
Just keep one function and decode & validate provider spec and validate secret there. Name it accordingly.